pkg/lintcontext/context.go

```diff
@@ -1,22 +1,64 @@
 package lintcontext
 
 import (
+	"encoding/json"
+
 	"golang.stackrox.io/kube-linter/internal/k8sutil"
+	"golang.stackrox.io/kube-linter/internal/stringutils"
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
 // ObjectMetadata is metadata about an object.
 type ObjectMetadata struct {
 	FilePath string
-	Raw      []byte
+	Raw      []byte `json:"-"`
 }
 
 // An Object references an object that is loaded from a YAML file.
 type Object struct {
 	Metadata  ObjectMetadata
-	K8sObject k8sutil.Object
+	K8sObject k8sutil.Object `json:"-"`
+}
+
+// K8sObjectInfo contains identifying information about k8s object.
+type K8sObjectInfo struct {
+	Namespace, Name, GroupVersionKind string
```

Comment:
(Sorry, I made this comment before, but looks like it somehow didn't get flushed into the review)

Just use `schema.GroupVersionKind` for the `GroupVersionKind` instead of `string`? Since it's JSON, we should make the output as structured as possible.

Reply:
Makes sense. Here's a sample of updated output

```json
    {
      "Diagnostic": {
        "Message": "container \"splunk\" has memory limit 0"
      },
      "Check": "unset-memory-requirements",
      "Remediation": "Set your container's memory requests and limits depending on its requirements. See https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits for more details.",
      "Object": {
        "Metadata": {
          "FilePath": "/home/misha/tmp/splunk.yaml"
        },
        "K8sObject": {
          "Namespace": "splunk",
          "Name": "splunk",
          "GroupVersionKind": {
            "Group": "apps",
            "Version": "v1",
            "Kind": "Deployment"
          }
        }
      }
    }
```